assert and type concepts (#650)
Codecademy/docs: content/python/concepts/assert/assert.md

---
Title: 'assert'
Description: 'A keyword for confirming the truthiness of a given statement'
Subjects:
  - 'Computer Science'
Tags:
  - 'Assert'
CatalogContent:
  - 'learn-python-3'
  - 'paths/computer-science'
---

'A keyword for confirming the truthiness of a given statement that is often used to facilitate development, document code and debug.'

## Syntax

```pseudo
assert some_condition, assert_message
```

> The `assert` keyword is followed by the statement containing the condition to be tested, such as validating a data type or membership within a given container, etc. An optional message can be added that will be printed when the assertion error is raised.

Reviewer: The "etc." sounds a bit abrupt here. I think it can be removed.
```suggestion
The `assert` keyword is followed by the statement containing the condition to be tested, such as validating a data type or membership within a given container. An optional message can be added that will be printed when the assertion error is raised.
```

---
Title: 'assert'
Description: 'A keyword for confirming the truthiness of a given statement'
Subjects:
  - 'Computer Science'
Tags:
  - 'Assert'
  - 'Debugging'
  - 'Development'
CatalogContent:
  - 'learn-python-3'
  - 'paths/computer-science'
---

> A keyword for confirming the truthiness of a given statement that is often used to facilitate development, document code and debug.

Reviewer 2: ```suggestion
A keyword for confirming the truthiness of a given statement that is often used to facilitate development, document code, and debug.
```

Author: This punctuation was intentional, the requirement of commas before every item in a list is a common misconception. Previously, I would always add these; however, it's really only required when joining independent clauses.

Reviewer 3: @sonnynomnom can we confirm? Full disclosure, I'm not too familiar with the ins-and-outs of the oxford comma but I assumed it was to be used all the time?

Reviewer 4: Yeah, I just checked again. I think for Codecademy content, the oxford comma is required according to the Curriculum Style Guide created by a bunch of our colleagues:

https://curriculum-documentation.codecademy.com/content-guidelines/editorial-pedagogy/#punctuation

## Syntax

```pseudo
assert some_condition, assert_message
```

The `assert` keyword is followed by the statement containing the condition to be tested, such as validating a data type or membership within a given container. An optional message can be added that will be printed when the assertion error is raised.

## Example

The `assert` keyword can be used to validate a value (or evaluate some other condition):

```py
fav_color = 'blue'

assert fav_color == 'red'
```

## Codebyte Example

> The following example uses `assert` to evaluate if 'mud' is within the following list `things_that_float`:

Reviewer 2: ```suggestion
The following example uses `assert` to evaluate if 'mud' is within the following `things_that_float` list:
```

Author: Not sure I see a benefit in readability by inverting the name and type

Reviewer 2: Alright! You can resolve this convo in that case 👍

---
Title: 'assert'

> Description: 'A keyword for confirming the truthiness of a given statement.'

Reviewer 3: My understanding is that the `Description` differs between term entries and concept entries (which is what this is). For concept entries, they should include (preferably begin with) the concept-by-name. For example:

```suggestion
Description: 'The assert keyword confirms the truthiness of a given statement.'
```
